### PR TITLE
Shell scripts refactoring

### PR DESCRIPTION
**Pull request by dzikowski**

### CODE REVIEW

File: e2e/__snapshots__/extendConfig.test.js.snap [hunk 1]
```diff
@@ -4806,7 +4806,7 @@ Object {
           "requiredPeerCount": 1,
         },
       ],
-      "privateDataEnabled": true,
+      "privateDataConfigFile": "collections/or-policy-chaincode.json",
```
Comment (Hejwo) [inline e2e/__snapshots__/extendConfig.test.js.snap:4809]: :+1:

File: generators/extend-config/configTransformers.js [hunk 2]
```diff
@@ -43,7 +43,7 @@ function transformChaincodesConfig(fabricVersion, chaincodes, transformedChannel
 
     const privateData = (chaincode.privateData || [])
       .map((d) => createPrivateCollectionConfig(fabricVersion, channel, d.name, d.orgNames));
-    const privateDataEnabled = privateData.length > 0;
```
Comment (Hejwo) [inline generators/extend-config/configTransformers.js:46]: :+1:

File: generators/setup-docker/templates/fabric-docker/commands-generated.sh [hunk 3]
```diff
@@ -16,23 +16,36 @@ function installChaincodes() {
...
-      <% if(!networkSettings.tls) { -%>
-      chaincodeInstall "cli.<%= org.domain %>" "<%= peer.fullAddress %>" "<%= chaincode.channel.name %>" "<%= chaincode.name %>" "<%= chaincode.version %>" "<%= chaincode.lang %>" "$CHAINCODES_BASE_DIR/<%= chaincode.directory %>" "<%= rootOrg.ordererHead.fullAddress %>"
-      <% } else { -%>
-      chaincodeInstallTls "cli.<%= org.domain %>" "<%= peer.fullAddress %>" "<%= chaincode.channel.name %>" "<%= chaincode.name %>" "<%= chaincode.version %>" "<%= chaincode.lang %>" "$CHAINCODES_BASE_DIR/<%= chaincode.directory %>" "<%= rootOrg.ordererHead.fullAddress %>" "crypto/orderer-tlscacerts/tlsca.<%= rootOrg.domain %>-cert.pem"
-      <% } -%>
-      <% })}) -%>
+      <% chaincode.channel.orgs.forEach(function (org) { -%>
+        <% org.peers.forEach(function (peer) { -%>
+          printHeadline "Installing '<%= chaincode.name %>' on <%= chaincode.channel.name %>/<%= org.name %>/<%= peer.name %>" "U1F60E"
+          chaincodeInstall \
```
Comment (Hejwo) [inline generators/setup-docker/templates/fabric-docker/commands-generated.sh:22]: hm... not sure about this one. `commands-generated.sh` tends to be big file. We also can install multiple chaincodes on multiple channels and peers. I would keep it as one liner as it's easier to reason about whole network setup that way.

File: generators/setup-docker/templates/fabric-docker/commands-generated.sh [hunk 4]
```diff
@@ -16,23 +16,36 @@ function installChaincodes() {
...
 
       printItalics "Instantiating chaincode '<%= chaincode.name %>' on channel '<%= chaincode.channel.name %>' as '<%= chaincode.instantiatingOrg.name %>'" "U1F618"
-      <% if(!networkSettings.tls) { -%>
-      chaincodeInstantiate "cli.<%= chaincode.instantiatingOrg.domain %>" "<%= chaincode.instantiatingOrg.headPeer.fullAddress %>" "<%= chaincode.channel.name %>" "<%= chaincode.name %>" "<%= chaincode.version %>" "<%= chaincode.lang %>" "$CHAINCODES_BASE_DIR/<%= chaincode.directory %>" "<%= rootOrg.ordererHead.fullAddress %>" '<%- chaincode.init %>' "<%- chaincode.endorsement %>" "<%= chaincode.privateDataEnabled %>"
-      <% } else { -%>
-      chaincodeInstantiateTls "cli.<%= chaincode.instantiatingOrg.domain %>" "<%= chaincode.instantiatingOrg.headPeer.fullAddress %>" "<%= chaincode.channel.name %>" "<%= chaincode.name %>" "<%= chaincode.version %>" "<%= chaincode.lang %>" "$CHAINCODES_BASE_DIR/<%= chaincode.directory %>" "<%= rootOrg.ordererHead.fullAddress %>" '<%- chaincode.init %>' "<%- chaincode.endorsement %>" "<%= chaincode.privateDataEnabled %>" "crypto/orderer-tlscacerts/tlsca.<%= rootOrg.domain %>-cert.pem"
-      <% } -%>
+      chaincodeInstantiate \
```
Comment (Hejwo) [inline generators/setup-docker/templates/fabric-docker/commands-generated.sh:36]: hm... not sure about this one. `commands-generated.sh` tends to be big file. We also can install instantiate chaincodes on multiple channels. I would keep it as one liner as it's easier to reason about whole network setup that way.

File: generators/setup-docker/templates/fabric-docker/commands-generated.sh [hunk 5]
```diff
@@ -51,76 +64,122 @@ function upgradeChaincode() {
...
-    chaincodeUpgradeTls "cli.<%= chaincode.instantiatingOrg.domain %>" "<%=  chaincode.instantiatingOrg.headPeer.fullAddress %>" "<%= chaincode.channel.name %>" "<%= chaincode.name %>" "$version" "<%= chaincode.lang %>" "$CHAINCODES_BASE_DIR/<%= chaincode.directory %>" "<%= rootOrg.ordererHead.fullAddress %>" '<%- chaincode.init %>' "<%- chaincode.endorsement %>" "<%= chaincode.privateDataEnabled %>" "crypto/orderer-tlscacerts/tlsca.<%= rootOrg.domain %>-cert.pem"
-    <% } %>
-  fi
+  <% chaincodes.forEach(function(chaincode) { -%>
+    if [ "$chaincodeName" = "<%= chaincode.name %>" ]; then
+      <%- include('commands-generated-node-build.sh.ejs', {chaincode: chaincode}); -%>
+      <% chaincode.channel.orgs.forEach(function (org) { -%>
+        <% org.peers.forEach(function (peer) { -%>
+          printHeadline "Installing '<%= chaincode.name %>' on <%= chaincode.channel.name %>/<%= org.name %>/<%= peer.name %>" "U1F60E"
+          chaincodeInstall \
```
Comment (Hejwo) [inline generators/setup-docker/templates/fabric-docker/commands-generated.sh:73]: same as above - i would reduce it. simply it's too much lines for commands-generated

File: generators/setup-docker/templates/fabric-docker/commands-generated.sh [hunk 6]
```diff
@@ -51,76 +64,122 @@ function upgradeChaincode() {
...
+            "<%= chaincode.lang %>" \
+            "$CHAINCODES_BASE_DIR/<%= chaincode.directory %>" \
+            "<%= rootOrg.ordererHead.fullAddress %>" \
+            "<%= !networkSettings.tls ? '' : 'crypto/orderer-tlscacerts/tlsca.' + rootOrg.domain+ '-cert.pem' %>"
+        <% }) -%>
+      <% }) -%>
+
+      printItalics "Upgrading as '<%= chaincode.instantiatingOrg.name %>'. '<%= chaincode.name %>' on channel '<%= chaincode.channel.name %>'" "U1F618"
+      chaincodeUpgrade \
+        "cli.<%= chaincode.instantiatingOrg.domain %>" \
```
Comment (Hejwo) [inline generators/setup-docker/templates/fabric-docker/commands-generated.sh:88]: same as above - i would reduce it. simply it's too much lines for commands-generated

**Review by Hejwo — CHANGES_REQUESTED** [review]

File: generators/setup-docker/templates/fabric-docker/commands-generated.sh [hunk 7]
```diff
@@ -51,76 +64,122 @@ function upgradeChaincode() {
...
 }
 
 function notifyOrgsAboutChannels() {
   printHeadline "Creating new channel config blocks" "U1F537"
-  <% channels.forEach(function(channel){  channel.orgs.forEach(function(org){ -%>
-createNewChannelUpdateTx "<%= channel.name %>" "<%= org.mspName %>" "<%= channel.profile.name %>" "$FABRICA_NETWORK_ROOT/fabric-config"  "$FABRICA_NETWORK_ROOT/fabric-config/config"
-  <% })}) %>
+  <% channels.forEach(function(channel){ -%>
+    <% channel.orgs.forEach(function(org){ -%>
+      createNewChannelUpdateTx \
```
Comment (Hejwo) [inline generators/setup-docker/templates/fabric-docker/commands-generated.sh:107]: same as above - i would reduce it.  
also - it wasn't so long anyways

File: generators/setup-docker/templates/fabric-docker/commands-generated.sh [hunk 8]
```diff
@@ -51,76 +64,122 @@ function upgradeChaincode() {
...
-  <% if(!networkSettings.tls) { -%>
-notifyOrgAboutNewChannel "<%= channel.name %>" "<%= org.mspName %>" "cli.<%= org.domain %>" "peer0.<%= org.domain %>" "<%= rootOrg.ordererHead.fullAddress %>"
-  <% } else { -%>
-notifyOrgAboutNewChannelTls "<%= channel.name %>" "<%= org.mspName %>" "cli.<%= org.domain %>" "peer0.<%= org.domain %>" "<%= rootOrg.ordererHead.fullAddress %>" "crypto/orderer-tlscacerts/tlsca.<%= rootOrg.domain %>-cert.pem"
-  <% } -%>
-  <% })}) %>
+  <% channels.forEach(function(channel){ -%>
+    <% channel.orgs.forEach(function(org){ -%>
+     <% if(!networkSettings.tls) { -%>
+       notifyOrgAboutNewChannel \
```
Comment (Hejwo) [inline generators/setup-docker/templates/fabric-docker/commands-generated.sh:120]: same as above - i would reduce it.
also - it wasn't so long anyways

File: generators/setup-docker/templates/fabric-docker/commands-generated.sh [hunk 9]
```diff
@@ -170,23 +229,24 @@ function installChannels() {
...
-    echo "docker rm failed, Check if all fabric dockers properly was deleted"
-  }
-  <% })})}) -%>
+  <% chaincodes.forEach(function(chaincode) { -%>
+    <% chaincode.channel.orgs.forEach(function (org) { -%>
+      <% org.peers.forEach(function (peer) { -%>
+        <% const chaincodeContainerName="dev-"+peer.address+"-"+chaincode.name+"-"+chaincode.version -%>
+        docker rm -f $(docker ps -a | grep <%= chaincodeContainerName %>-* | awk '{print $1}') || {
+          echo "docker rm failed, Check if all fabric dockers properly was deleted"
+        }
```
Comment (Hejwo) [inline generators/setup-docker/templates/fabric-docker/commands-generated.sh:243]: I know that it's pretty old code, but... all of docker commands besides this one are wrapped into much shorter utility scripts. maybe we could wrap this one also?

File: generators/setup-docker/templates/fabric-docker/docker-compose.yaml [hunk 10]
```diff
@@ -106,7 +106,9 @@ services:
       <%_ } _%>
       <%_ chaincodes.forEach(function(chaincode) { _%>
       - "$CHAINCODES_BASE_DIR/<%= chaincode.directory %>/:/var/hyperledger/cli/<%= chaincode.name %>/"
-      - "../fabric-config/collections/<%= chaincode.name %>.json:/var/hyperledger/cli/collections/<%= chaincode.name %>.json"
+      <%_ if(chaincode.privateDataConfigFile) { _%>
```
Comment (Hejwo) [inline generators/setup-docker/templates/fabric-docker/docker-compose.yaml:109]: hm... didn't knew `_%>`. what it does?

File: generators/setup-docker/templates/fabric-docker/scripts/chaincode-functions.sh [hunk 11]
```diff
@@ -12,14 +12,26 @@ function chaincodeInstall() {
...
-  echo "Installing chaincode on $CHANNEL_NAME..."
+  local CA_CERT
+  local CA_CERT_PARAMS
+
+  if [ -n "$9" ]; then
+    CA_CERT="/var/hyperledger/cli/$9"
+    CA_CERT_PARAMS=(--tls --cafile "$CA_CERT")
+  else
+    CA_CERT="<none>"
+    CA_CERT_PARAMS=()
```
Comment (Hejwo) [inline generators/setup-docker/templates/fabric-docker/scripts/chaincode-functions.sh:23]: :+1:

File: generators/setup-docker/templates/fabric-docker/docker-compose.yaml (hunk 10 above)
Comment (dzikowski) [reply generators/setup-docker/templates/fabric-docker/docker-compose.yaml:109]: see the docs: https://ejs.co/

`_%>` trims the following white spaces. this is probably some legacy, I would use `<% xxx -%>`

**Review by Hejwo — APPROVED** [review]

**Pull request merged**